Sources/MobileSdk/CredentialPack.swift

```diff
@@ -1,53 +1,78 @@
+        self.credentials.append(ParsedCredential.newJwtVcJson(jwtVc: jwtVc))
+        return self.credentials
     }
 
-    public func addMDoc(mdocBase64: String, keyAlias: String = UUID().uuidString) throws -> [Credential]? {
-        let mdocData = Data(base64Encoded: mdocBase64)!
-        let credential = MDoc(fromMDoc: mdocData, namespaces: [:], keyAlias: keyAlias)!
-        self.credentials.append(credential)
+    /// Add a JsonVc to the CredentialPack.
+    public func addJsonVc(jsonVc: JsonVc) -> [ParsedCredential] {
+        self.credentials.append(ParsedCredential.newLdpVc(jsonVc: jsonVc))
         return self.credentials
     }
 
-    public func get(keys: [String]) -> [String: [String: GenericJSON]] {
-        var values: [String: [String: GenericJSON]] = [:]
-        for cred in self.credentials {
-            values[cred.id] = cred.get(keys: keys)
-        }
-
-        return values
+    /// Add an Mdoc to the CredentialPack.
+    public func addMDoc(mdoc: Mdoc) -> [ParsedCredential] {
+        self.credentials.append(ParsedCredential.newMsoMdoc(mdoc: mdoc))
+        return self.credentials
     }
 
-    public func get(credentialsIds: [String]) -> [Credential] {
-        return self.credentials.filter { credentialsIds.contains($0.id) }
+    /// Find credential claims from all credentials in this CredentialPack.
+    public func findCredentialClaims(claimNames: [String]) -> [Uuid: [String: GenericJSON]] {
+        Dictionary(uniqueKeysWithValues: self.list()
+            .map { credential in
+                var claims: [String: GenericJSON]
+                if let mdoc = credential.asMsoMdoc() {
+                    claims = mdoc.jsonEncodedDetails(containing: claimNames)
+                } else if let jwtVc = credential.asJwtVc() {
+                    claims = jwtVc.credentialClaims(containing: claimNames)
+                } else if let jsonVc = credential.asJsonVc() {
+                    claims = jsonVc.credentialClaims(containing: claimNames)
+                } else {
+                    var type: String
+                    do {
+                        type = try credential.intoGenericForm().type
+                    } catch {
+                        type = "unknown"
+                    }
+                    print("unsupported credential type: \(type)")
+                    claims = [:]
+                }
+                return (credential.id(), claims)
+            })
     }
 
-    public func get(credentialId: String) -> Credential? {
-        if let credential = self.credentials.first(where: { $0.id == credentialId }) {
-           return credential
-        } else {
-           return nil
+    /// Get credentials by id.
+    public func get(credentialsIds: [Uuid]) -> [ParsedCredential] {
+        return self.credentials.filter {
+            credentialsIds.contains($0.id())
         }
     }
+
+    /// Get a credential by id.
+    public func get(credentialId: Uuid) -> ParsedCredential? {
```

Comment:
I'm a bit weirded out by having two functions with the same name and just an argument name that differs by one letter, but that might also just be a Swift convention, and I don't know Swift that well.

Reply:
I think this is a fairly normal Swift convention, but maybe @todd-spruceid or @Juliano1612 can weigh in.

Reply:
Yeah, that's common. You only can't have multiple methods with the same signature.